inc/saiport.h

```diff
@@ -2108,6 +2120,233 @@ typedef sai_status_t (*sai_clear_port_pool_stats_fn)(
+     *
+     * List of port serdes TX fir precursor1 tap-filter values.
+     * The values are of type sai_u32_list_t where the count is number lanes in
+     * a port and the list specifies list of values to be applied to each lane.
+     *
+     * @type sai_u32_list_t
+     * @flags CREATE_AND_SET
+     * @default internal
+     */
+    SAI_PORT_SERDES_ATTR_TX_FIR_PRE1,
+
+    /**
+     * @brief Port serdes control TX FIR PRE2 filter
+     *
+     * List of port serdes TX fir precursor2 tap-filter values.
+     * The values are of type sai_u32_list_t where the count is number lanes in
+     * a port and the list specifies list of values to be applied to each lane.
+     *
+     * @type sai_u32_list_t
+     * @flags CREATE_AND_SET
+     * @default internal
+     */
+    SAI_PORT_SERDES_ATTR_TX_FIR_PRE2,
+
+    /**
+     * @brief Port serdes control TX FIR PRE3 filter
+     *
+     * List of port serdes TX fir precursor3 tap-filter values.
+     * The values are of type sai_u32_list_t where the count is number lanes in
+     * a port and the list specifies list of values to be applied to each lane.
+     *
+     * @type sai_u32_list_t
+     * @flags CREATE_AND_SET
+     * @default internal
+     */
+    SAI_PORT_SERDES_ATTR_TX_FIR_PRE3,
+
+    /**
+     * @brief Port serdes control TX FIR MAIN filter
+     *
+     * List of port serdes TX fir maincursor tap-filter values.
+     * The values are of type sai_u32_list_t where the count is number lanes in
+     * a port and the list specifies list of values to be applied to each lane.
+     *
+     * @type sai_u32_list_t
+     * @flags CREATE_AND_SET
+     * @default internal
+     */
+    SAI_PORT_SERDES_ATTR_TX_FIR_MAIN,
+
+    /**
+     * @brief Port serdes control TX FIR POST1 filter
+     *
+     * List of port serdes TX fir postcursor1 tap-filter values.
+     * The values are of type sai_u32_list_t where the count is number lanes in
+     * a port and the list specifies list of values to be applied to each lane.
+     *
+     * @type sai_u32_list_t
+     * @flags CREATE_AND_SET
+     * @default internal
+     */
+    SAI_PORT_SERDES_ATTR_TX_FIR_POST1,
+
+    /**
+     * @brief Port serdes control TX FIR POST2 filter
+     *
+     * List of port serdes TX fir postcursor2 tap-filter values.
+     * The values are of type sai_u32_list_t where the count is number lanes in
+     * a port and the list specifies list of values to be applied to each lane.
+     *
+     * @type sai_u32_list_t
+     * @flags CREATE_AND_SET
+     * @default internal
+     */
+    SAI_PORT_SERDES_ATTR_TX_FIR_POST2,
+
+    /**
+     * @brief Port serdes control TX FIR POST3 filter
+     *
+     * List of port serdes TX fir postcursor3 tap-filter values.
+     * The values are of type sai_u32_list_t where the count is number lanes in
+     * a port and the list specifies list of values to be applied to each lane.
+     *
+     * @type sai_u32_list_t
+     * @flags CREATE_AND_SET
+     * @default internal
```

Comment:
i would propose to use "empty" value here, if this port_serdes object is created are those list populated internally ? or they empty and user must set values ?

Reply:
Hi kamil,
These values can be internally populated from ASIC specific files (for e.g config.bcm). That's why these were introduced earlier for pre-emphasis(refer existing SAI_PORT_ATTR_SERDES_PREEMPHASIS)

Reply:
Hi dgsudharsan,
These values can be populated from ASIC specific files (for e.g config.bcm) indeed.
If applications want to SET the better values in runtime, the flags should be CREATE_AND_SET, am I right?

@dgsudharsan

Reply:
Hi dgsudharsan,
Defining the type of these port_serdes_attr (pre1, pre2...) as **sai_s32_list_t** may be better. Because they are possible to be negative values.